esphome/core/progmem.h

```diff
@@ -1,10 +1,10 @@
 #pragma once
 
 // Platform-agnostic macros for PROGMEM string handling
-// On ESP32 (both Arduino and IDF): Use plain strings (no PROGMEM)
+// On ESP32 (both Arduino and IDF) or HOST: Use plain strings (no PROGMEM)
 // On ESP8266/Arduino: Use Arduino's F() macro for PROGMEM strings
 
-#ifdef USE_ESP32
+#if defined(USE_ESP32) or defined(USE_HOST)
```

Comment:
AFAIK it is *only* ESP8266 that can't read bytes from flash memory.

```suggestion
#ifndef USE_ESP8266
```